buildkite/src/Jobs/CheckDhall/Spec.dhall

```diff
@@ -0,0 +1,7 @@
+let JobSpec = ../../Pipeline/JobSpec.dhall
+
+in
+JobSpec::{
+  dirtyWhen = "^buildkite/(.*\\.dhall\\\$|Makefile\\\$|scripts/generate-jobs\\.sh\\\$)",
```

Comment:
Isn't there a single quote in dhall that you can use? The escaping here is ugly and hard to read.

Reply:
As far as I can tell, the only way to build strings that ignore escaping is via and escaped interpolation `''${ }` which also, unfortunately, includes the `${ }`.

I'll open an issue to track this because I don't want to block the PR on this point, but I definitely agree we should come up with some better way of building these regular expressions.

Reply:
Tracked on #5001